Directive "changelog" returned non-Node object

Without a GitHub token, the "no changelog" page is built successfully.
When passing a token via the SPHINX_GITHUB_CHANGELOG_TOKEN environment variable the docs fail to build with the following error:

```
# Sphinx version: 3.5.3
# Python version: 3.9.0 (CPython)
# Docutils version: 0.16 release
# Jinja2 version: 2.11.3
# Last messages:
#   Running Sphinx v3.5.3
#   loading pickled environment...
#   done
#   building [mo]: targets for 0 po files that are out of date
#   building [html]: targets for 10 source files that are out of date
#   updating environment:
#   0 added, 2 changed, 0 removed
#   reading sources... [ 50%] changelog
# Loaded extensions:
#   sphinx.ext.mathjax (3.5.3) from \path\to\project\venv\lib\site-packages\sphinx\ext\mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from \path\to\project\venv\lib\site-packages\sphinxcontrib\applehelp\__init__.py
#   sphinxcontrib.devhelp (1.0.2) from \path\to\project\venv\lib\site-packages\sphinxcontrib\devhelp\__init__.py
#   sphinxcontrib.htmlhelp (1.0.3) from \path\to\project\venv\lib\site-packages\sphinxcontrib\htmlhelp\__init__.py
#   sphinxcontrib.serializinghtml (1.1.4) from \path\to\project\venv\lib\site-packages\sphinxcontrib\serializinghtml\__init__.py
#   sphinxcontrib.qthelp (1.0.3) from \path\to\project\venv\lib\site-packages\sphinxcontrib\qthelp\__init__.py
#   alabaster (0.7.12) from \path\to\project\venv\lib\site-packages\alabaster\__init__.py
#   sphinx_rtd_theme (unknown version) from \path\to\project\venv\lib\site-packages\sphinx_rtd_theme\__init__.py
#   sphinx.ext.autodoc.type_comment (3.5.3) from \path\to\project\venv\lib\site-packages\sphinx\ext\autodoc\type_comment.py
#   sphinx.ext.autodoc (3.5.3) from \path\to\project\venv\lib\site-packages\sphinx\ext\autodoc\__init__.py
#   sphinx_github_changelog (1.0.2) from \path\to\project\venv\lib\site-packages\sphinx_github_changelog\__init__.py
Traceback (most recent call last):
  File "\path\to\project\venv\lib\site-packages\sphinx\cmd\build.py", line 280, in build_main
    app.build(args.force_all, filenames)
  File "\path\to\project\venv\lib\site-packages\sphinx\application.py", line 352, in build
    self.builder.build_update()
  File "\path\to\project\venv\lib\site-packages\sphinx\builders\__init__.py", line 296, in build_update
    self.build(to_build,
  File "\path\to\project\venv\lib\site-packages\sphinx\builders\__init__.py", line 310, in build
    updated_docnames = set(self.read())
  File "\path\to\project\venv\lib\site-packages\sphinx\builders\__init__.py", line 417, in read
    self._read_serial(docnames)
  File "\path\to\project\venv\lib\site-packages\sphinx\builders\__init__.py", line 438, in _read_serial
    self.read_doc(docname)
  File "\path\to\project\venv\lib\site-packages\sphinx\builders\__init__.py", line 478, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "\path\to\project\venv\lib\site-packages\sphinx\io.py", line 221, in read_doc
    pub.publish()
  File "\path\to\project\venv\lib\site-packages\docutils\core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "\path\to\project\venv\lib\site-packages\sphinx\io.py", line 126, in read
    self.parse()
  File "\path\to\project\venv\lib\site-packages\docutils\readers\__init__.py", line 77, in parse
    self.parser.parse(self.input, document)
  File "\path\to\project\venv\lib\site-packages\sphinx\parsers.py", line 104, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "\path\to\project\venv\lib\site-packages\docutils\statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "\path\to\project\venv\lib\site-packages\docutils\statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "\path\to\project\venv\lib\site-packages\docutils\statemachine.py", line 241, in run
    context, next_state, result = self.check_line(
  File "\path\to\project\venv\lib\site-packages\docutils\statemachine.py", line 459, in check_line
    return method(match, context, next_state)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 2096, in directive
    return self.run_directive(
  File "\path\to\project\venv\lib\site-packages\docutils\parsers\rst\states.py", line 2155, in run_directive
    assert isinstance(result[i], nodes.Node), \
AssertionError: Directive "changelog" returned non-Node object (index 0): []

```

--- a/sphinx_github_changelog/changelog.py
+++ b/sphinx_github_changelog/changelog.py
@@ -1,4 +1,4 @@
-from typing import Any, Dict, Iterable, List, Optional
+from typing import Any, Dict, Iterable, Optional
 
 import requests
 from docutils import nodes
@@ -45,11 +45,11 @@ def compute_changelog(
 
     pypi_name = extract_pypi_package_name(url=options.get("pypi"))
 
-    result_nodes: List[nodes.Node] = []
-    for release in releases:
-        result_nodes.append(nodes_for_release(release=release, pypi_name=pypi_name))
+    result_nodes = (
+        node_for_release(release=release, pypi_name=pypi_name) for release in releases
+    )
 
-    return result_nodes
+    return [n for n in result_nodes if n is not None]
 
 
 def no_token(changelog_url: Optional[str]) -> Iterable[nodes.Node]:
@@ -97,11 +97,11 @@ def extract_pypi_package_name(url: Optional[str]) -> Optional[str]:
     return stripped_url[len(prefix) :]  # noqa
 
 
-def nodes_for_release(
+def node_for_release(
     release: Dict[str, Any], pypi_name: Optional[str] = None
-) -> Iterable[nodes.Node]:
+) -> Optional[nodes.Node]:
     if release["isDraft"]:
-        return []  # For now, draft releases are excluded
+        return None  # For now, draft releases are excluded
 
     tag = release["tagName"]
     title = release["name"]

--- a/tests/unit/test_changelog.py
+++ b/tests/unit/test_changelog.py
@@ -128,9 +128,9 @@ def test_extract_pypi_package_name_error():
         changelog.extract_pypi_package_name("https://example.com")
 
 
-def test_nodes_for_release_no_pypi(release_dict):
+def test_node_for_release_no_pypi(release_dict):
     assert node_to_string(
-        changelog.nodes_for_release(release=release_dict, pypi_name=None)
+        changelog.node_for_release(release=release_dict, pypi_name=None)
     ) == canonicalize(
         """
         <section ids="release-1-0-0">
@@ -146,16 +146,16 @@ def test_nodes_for_release_no_pypi(release_dict):
     )
 
 
-def test_nodes_for_release_title_tag(release_dict):
+def test_node_for_release_title_tag(release_dict):
     release_dict["name"] = "Bla 1.0.0"
     assert "<title>Bla 1.0.0</title>" in node_to_string(
-        changelog.nodes_for_release(release=release_dict, pypi_name=None)
+        changelog.node_for_release(release=release_dict, pypi_name=None)
     )
 
 
-def test_nodes_for_release_title_pypy(release_dict):
+def test_node_for_release_title_pypy(release_dict):
     value = node_to_string(
-        changelog.nodes_for_release(release=release_dict, pypi_name="foo")
+        changelog.node_for_release(release=release_dict, pypi_name="foo")
     )
 
     assert (
@@ -164,9 +164,9 @@ def test_nodes_for_release_title_pypy(release_dict):
     )
 
 
-def test_nodes_for_release_draft(release_dict):
+def test_node_for_release_draft(release_dict):
     release_dict["isDraft"] = True
-    assert changelog.nodes_for_release(release=release_dict, pypi_name="foo") == []
+    assert changelog.node_for_release(release=release_dict, pypi_name="foo") is None
 
 
 def test_extract_releases(github_payload, release_dict, mocker):

